staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -225,6 +227,7 @@ func (m *Broadcaster) distribute(event Event) {
 			case w.result <- event:
 			case <-w.stopped:
 			default: // Don't block if the event can't be queued.
+				glog.Warningf("dropping event: %v", event)
```

Comment:
the intent of this is to limit excessive events. I don't think trading them for excessive logs is desired. cc @derekwaynecarr

Reply:
I recommend incrementing a counter (e.g. atomic.AddInt64), and either periodically logging it or exposing it as a metric instead of logging.

Reply:
@lavalamp this may require a new design for the architecture, I'm not sure if I get your point clearly. If do, we would have some things to solve:
1. to keep a counter we would need a cache, how long would data be held in this cache?
2. if the same event distribute successfully, would the event be erased from the cache?
just some ideas about your point, would be nice to hear your feedback :)

Reply:
@lavalamp as now, the dropping event occurs quite rarely. I think just logging is enough for admins to check wether there is dropping event. WDYT?

Reply:
> @lavalamp as now, the dropping event occurs quite rarely. I think just logging is enough for admins to check wether there is dropping event.

it may happen rarely in your environment, but the reason this was added was to allow shedding events in high-frequency cases. Logging each event in full is not appropriate here.